### PR TITLE
tests: Accept more IPs in login tests

### LINKED ISSUE #14874: TestLogin.{TestBasic, TestLogging} both expect the test user to be connected in remote machines
We need to adjust so that the tests can run in fedora CI as well, where the test suite is spawned with `--machine localhost:22` and `--browser localhost:9090`, thus the test user is connected to the local browser.

See an example of the test failure.
https://jenkins-continuous-infra.apps.ci.centos.org/job/fedora-f33-build-pipeline/216/artifact/package-tests/logs/FAIL-str_verify.log
TestLogin-testLogging-fedora-32-localhost-22-FAIL

The expected IP is 172.27.0.2 but the test fails since the last-login-host shows ::1 which points out that the connection is local.

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
We run the login tests not only on our own test VMs but also as part of
distro package validation testing.  Stop hardcoding the IP we use in our
test framework.

Fixes #14874

### CODE REVIEW

File: test/verify/check-login
Comment (martinpitt) [inline test/verify/check-login:107]: Could we keep this IPv4 compatible? For example, with `\d+[.:]\d+` . I'm not sure if the machines in Fedora/RHEL gating speak IPv6.

Comment (allisonkarlitskaya) [reply test/verify/check-login:107]: As per the screenshot in #14874, Fedora ends up being `::1`.  RHEL is another question.

Comment (martinpitt) [reply test/verify/check-login:107]: ah, that wouldn't work with `::1`. Maybe `(\d+\.\d+|::)`?

Comment (allisonkarlitskaya) [reply test/verify/check-login]: Took your suggestion, though.  Will make this more robust against all kinds of possible environments.  Thanks!

**Review by martinpitt — APPROVED** [review]
Cheers! LGTM, if tests agree.

**Comment by martinpitt** [comment]
Hmm, two unit tests failed due to timeout; but these are still the old ones before my changes in PR #14852. But gcc succeeded, so I suppose it was just really bad luck. I uploaded the new containers now, but these also need the corresponding changes to containers/unit-tests/, so I'm afraid you'll need to rebase.

**Review by martinpitt — DISMISSED** [review]
Still blocked on our integration tests being AWOL, but LGTM. Danke!

**Review by martinpitt — APPROVED** [review]

**Pull request merged**
